lang/cs/Org.Apache.REEF.Common/Evaluator/DriverInformation.cs

```diff
@@ -89,48 +90,73 @@ public string NameServerId
 
         public static DriverInformation GetDriverInformationFromHttp(Uri queryUri)
         {
-            HttpWebRequest request = (HttpWebRequest)WebRequest.Create(queryUri);
-            request.AllowAutoRedirect = false;
-            request.KeepAlive = false;
-            request.ContentType = "text/html";
-
-            string driverInfomation;
-            AvroDriverInfo info = null;
-            try
+            while (queryUri != null)
             {
-                using (HttpWebResponse webResponse = (HttpWebResponse)request.GetResponse())
+                HttpWebRequest request = (HttpWebRequest)WebRequest.Create(queryUri);
+                request.AllowAutoRedirect = true;
+                request.KeepAlive = false;
+                request.ContentType = "text/html";
+
+                queryUri = null;
+                try
                 {
-                    Stream stream = webResponse.GetResponseStream();
-                    if (stream == null)
-                    {
-                        return null;
-                    }
-                    using (StreamReader streamReader = new StreamReader(stream, Encoding.UTF8))
+                    using (HttpWebResponse webResponse = (HttpWebResponse)request.GetResponse())
                     {
-                        driverInfomation = streamReader.ReadToEnd();
-                        LOGGER.Log(Level.Verbose, "Http response line: " + driverInfomation);
-                        info = AvroJsonSerializer<AvroDriverInfo>.FromString(driverInfomation);
+                        var refresh = webResponse.Headers.AllKeys.FirstOrDefault(k => k.Equals("refresh", StringComparison.OrdinalIgnoreCase));
+
+                        if (refresh != null)
```

Comment:
I wasn't able to find one that handles meta-refresh, though YARN does have a fix for this in 2.7.0 to respond with a 302 instead of what we have now.